org.osgi.test.assertj.framework/src/main/java/org/osgi/test/assertj/dictionary/DictionaryAssert.java

```diff
@@ -45,6 +46,10 @@ protected DictionaryAssert(Map<KEY, VALUE> actual) {
 		return new DictionaryAssert<>(actual);
 	}
 
+	public static <K, V> DictionaryAssert<K, V> assertThat(Hashtable<K, V> actual) {
+		return new DictionaryAssert<>((Dictionary<K, V>) actual);
```

Comment:
Why would you not use MapAssert for Hashtable? Since Dictionaries.asMap will see that Hashtable is-a Map and return it cast to Map. So there is no need for all that since we can just do

```java
public static <K, V> MapAssert<K, V> assertThat(Hashtable<K, V> actual) {
		return new assertThat((Map<K,V>) actual);
}
```

Reply:
Because DictionaryAssert has all the same methods that MapAssert does, but also the extra methods that you added which take a Dictionary for comparison. I thought these might be useful. In fact,  we should probably add variants that take a Hashtable to resolve the ambiguity there too.

Reply:
At a basic level, I don't think we should do anything for Hashtable. People can cast to Map or DIctionary as they wish. With Maps, people should use Map as the variable type: `Map map = new HashMap();` When using Hashtables, people really ought to declare the variable type as Dictionary (or Map): `Dictionary dict = new Hashtable();`. This then removes all ambiguity.

> but also the extra methods that you added which take a Dictionary for comparison.

Sure, but what about all the people that have maps that want to compare against dictionaries? We have Dictionaries.asDictionary for them. We cannot solve all combinations and I am not sure we should try.

Reply:
I am convinced. I'll drop the hashtable patch from this PR.